api/client/login.go

```diff
@@ -0,0 +1,118 @@
+package client
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+
+	"github.com/appcelerator/amp/api/rpc/oauth"
+	"github.com/fatih/color"
+	"github.com/howeyc/gopass"
+	"golang.org/x/net/context"
+)
+
+func getUsername() (username string) {
+	color.Set(color.FgMagenta, color.Bold)
+	fmt.Println("Login using github")
+	fmt.Print("github username or email: ")
+	color.Unset()
+	fmt.Scanln(&username)
+	return
+}
+
+func getPassword() (password string, err error) {
+	color.Set(color.FgMagenta, color.Bold)
+	defer color.Unset()
+	fmt.Print("github password: ")
+	pw, err := gopass.GetPasswd()
+	if err != nil {
+		return
+	}
+	password = string(pw)
+	fmt.Println("")
+	return
+}
+
+func getOTP() (otp string, err error) {
+	color.Set(color.FgBlue)
+	defer color.Unset()
+	fmt.Println("Two Factor Authentication required")
+	fmt.Print("authentication code: ")
+	otpraw, err := gopass.GetPasswd()
+	if err != nil {
+		return
+	}
+	fmt.Println("")
+	otp = string(otpraw)
+	return
+}
+
+func (a *AMP) githubOauth(username, password, otp string) (lastEight string, err error) {
+	a.connect()
+	defer a.disconnect()
+	c := oauth.NewGithubClient(a.Conn)
+	authReply, err := c.Create(context.Background(), &oauth.AuthRequest{
+		Username: username,
+		Password: password,
+		Otp:      otp,
+	})
+	if err != nil {
+		return
+	}
+	lastEight = authReply.SessionKey
+	return
+}
+
+func basicLogin(a *AMP) (username, password, lastEight string, err error) {
+	username = getUsername()
+	password, err = getPassword()
+	if err != nil {
+		return
+	}
+	lastEight, err = a.githubOauth(username, password, "")
+	if err != nil {
+		badCredError := regexp.MustCompile("Bad credentials")
+		if badCredError.MatchString(err.Error()) {
+			return basicLogin(a)
+		}
+	}
+	return
+}
+
+func otpLogin(a *AMP, username, password string) (lastEight string, err error) {
+	otp, err := getOTP()
+	if err != nil {
+		return
+	}
+	lastEight, err = a.githubOauth(username, password, otp)
+	if err != nil {
+		otpError := regexp.MustCompile("Must specify two-factor authentication OTP code")
+		if otpError.MatchString(err.Error()) {
+			return otpLogin(a, username, password)
+		}
+	}
+	return
+}
+
+// Login creates a github access token and store it in your config file to authenticate further commands
+func (a *AMP) Login() {
+	color.Set(color.FgRed)
+	defer color.Unset()
+	username, password, lastEight, err := basicLogin(a)
+	if err != nil {
+		otpError := regexp.MustCompile("Must specify two-factor authentication OTP code")
+		if otpError.MatchString(err.Error()) {
+			lastEight, err = otpLogin(a, username, password)
+		}
+		if err != nil {
+			fmt.Println(err)
+			os.Exit(1)
+		}
+	}
+
+	a.Configuration.Github = lastEight
+	a.Configuration.Save()
+
+	color.Set(color.FgCyan)
+	fmt.Println("token saved")
```

Comment:
Nitpicking on a great PR ... but can you change this to something like "Logged in" or "<user> is now logged in" and on failure, something to correspond, like "Login failed for <user>"...